server/src/com/cloud/api/ApiServer.java

```diff
@@ -958,6 +959,9 @@ private boolean commandAvailable(final InetAddress remoteAddress, final String c
         } catch (final RequestLimitException ex) {
             s_logger.debug(ex.getMessage());
             throw new ServerApiException(ApiErrorCode.API_LIMIT_EXCEED, ex.getMessage());
+        }  catch (final UnavailableCommandException ex) {
+            s_logger.debug(ex.getMessage());
+            throw new ServerApiException(ApiErrorCode.UNSUPPORTED_ACTION_ERROR, ex.getMessage());
```

Comment:
Why not throw ServerApiException with suitable http error code so as to not cause log out in the UI. Simply handle permission denied exception here? (or the line below)

Reply:
@rhtyd I'm not sure I understand your suggestion. The Http error code is specified here, in line 964). The `ServerApiException` is caught on `ApiServlet` which generates the HTTP response.